### PR TITLE
Add a ivf reader

### PR DESCRIPTION
**Pull request by lu-zero**

### CODE REVIEW

File: ivf/src/lib.rs [hunk 1]
```diff
@@ -30,3 +30,54 @@ pub fn write_ivf_frame(
...
+
+  let mut _signature = [0u8; 4];
+  let mut tag = [0u8; 4];
+
+  br.read_bytes(&mut _signature)?;
+  let _v0: u16 = br.read(16)?;
+  let _v1: u16 = br.read(16)?;
+  br.read_bytes(&mut tag)?;
+
+  // log::debug!("sign {:?} version {} {} tag {:?}", &signature, v0, v1, &tag);
```
Comment (tdaede) [inline ivf/src/lib.rs]: drop?

File: ivf/src/lib.rs [hunk 2]
```diff
@@ -30,3 +30,54 @@ pub fn write_ivf_frame(
...
+  pub timebase_den: u32
+}
+
+pub fn read_header(r: &mut dyn io::Read) -> io::Result<Header> {
+  let mut br = BitReader::endian(r, LittleEndian);
+
+  let mut _signature = [0u8; 4];
+  let mut tag = [0u8; 4];
+
+  br.read_bytes(&mut _signature)?;
```
Comment (tdaede) [inline ivf/src/lib.rs]: Do you want to actually verify the signature?

File: ivf/src/lib.rs (hunk 1 above)
Comment (lu-zero) [reply ivf/src/lib.rs]: Sure

File: ivf/src/lib.rs (hunk 2 above)
Comment (lu-zero) [reply ivf/src/lib.rs]: Probably it is a good idea.

**Pull request merged**